migration/migration_pkg_test.go

```diff
@@ -25,3 +26,54 @@ func TestRevision0000(t *testing.T) {
 		require.EqualError(t, testDB.View(revision0000), "initial DB should be empty")
 	})
 }
+
+func TestRelocateContractStorageRootKeys(t *testing.T) {
+	testDB := pebble.NewMemTest()
+	t.Cleanup(func() {
+		require.NoError(t, testDB.Close())
+	})
+
+	txn := testDB.NewTransaction(true)
+
+	numberOfContracts := 5
+
+	// Populate the database with entries in the old location.
+	for i := 0; i < numberOfContracts; i++ {
+		exampleBytes := new(felt.Felt).SetUint64(uint64(i)).Bytes()
+		// Use exampleBytes for the key suffix (the contract address) and the value.
+		err := txn.Set(db.Unused.Key(exampleBytes[:]), exampleBytes[:])
+		require.NoError(t, err)
+	}
+
+	require.NoError(t, relocateContractStorageRootKeys(txn))
+
+	it, err := txn.NewIterator()
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		require.NoError(t, it.Close())
+	})
+	// Position the iterator so it is valid.
+	require.True(t, it.Seek(nil), "db should be nonempty")
+
+	// Each root-key entry should have been moved to its new location
+	// and the old entry should not exist.
+	for i := 0; i < numberOfContracts; i++ {
+		require.True(t, it.Valid(), "iterator is invalid, should be valid")
+
+		exampleBytes := new(felt.Felt).SetUint64(uint64(i)).Bytes()
+
+		// New entry exists.
+		found := it.Seek(db.ContractStorage.Key(exampleBytes[:]))
```

Comment:
I think Seek can return true even if the key that you are seeking to doesnt exist. So we need to verify that it.Key() is what we expect.

or we can straight up use txn.Get instead of an iterator.

Reply:
Yes. However, because this is a test and we are only iterating `numberOfContracts` times, we know the key has to exist. That's why we require `found` to be true.

Reply:
Actually, just took another look and `txn.Get` is a lot simpler than the iterator. Will use that. Thanks!